Library won't compile with -pedantic flag: `error: call of overloaded 'requestFrom(uint8_t, int, uint8_t)' is ambiguous`

- Arduino board:  Arduino Uno (Rev 3)

- Arduino IDE version (found in Arduino -> About Arduino menu): Arduino.mk 1.6.0

- List the steps to reproduce the problem below (if possible attach a sketch or
  copy the sketch code in too): Simply include the library and compile with `CXXFLAGS += -pedantic`

Output:

```
Arduino.mk Configuration:
- [AUTODETECTED]       CURRENT_OS = MAC 
- [USER]               ARDUINO_DIR = /Applications/Arduino.app/Contents/Java 
- [USER]               ARDMK_DIR = /usr/local/Cellar/arduino-mk/1.6.0 
- [AUTODETECTED]       ARDUINO_VERSION = 189 
- [DEFAULT]            ARCHITECTURE = avr 
- [DEFAULT]            ARDMK_VENDOR = arduino 
- [AUTODETECTED]       ARDUINO_PREFERENCES_PATH = /home/x/Library/Arduino15/preferences.txt 
- [AUTODETECTED]       ARDUINO_SKETCHBOOK = /home/x/Projects/Arduino (from arduino preferences file)
- [USER]               AVR_TOOLS_DIR = /usr/local 
- [COMPUTED]           ARDUINO_LIB_PATH = /Applications/Arduino.app/Contents/Java/libraries (from ARDUINO_DIR)
- [COMPUTED]           ARDUINO_PLATFORM_LIB_PATH = /Applications/Arduino.app/Contents/Java/hardware/arduino/avr/libraries (from ARDUINO_DIR)
- [COMPUTED]           ARDUINO_VAR_PATH = /Applications/Arduino.app/Contents/Java/hardware/arduino/avr/variants (from ARDUINO_DIR)
- [COMPUTED]           BOARDS_TXT = /Applications/Arduino.app/Contents/Java/hardware/arduino/avr/boards.txt (from ARDUINO_DIR)
- [USER]               USER_LIB_PATH = /home/x/Projects/y/z/software/arduino_uno/dragonfly//lib 
- [DEFAULT]            PRE_BUILD_HOOK = pre-build-hook.sh 
- [USER]               BOARD_TAG = uno 
- [COMPUTED]           CORE = arduino (from build.core)
- [COMPUTED]           VARIANT = standard (from build.variant)
- [USER]               OBJDIR = /home/x/Projects/y/z/software/arduino_uno/dragonfly//bin/uno/dragonfly 
- [COMPUTED]           ARDUINO_CORE_PATH = /Applications/Arduino.app/Contents/Java/hardware/arduino/avr/cores/arduino (from ARDUINO_DIR, BOARD_TAG and boards.txt)
- [USER]               MONITOR_BAUDRATE = 115200 
- [DEFAULT]            OPTIMIZATION_LEVEL = s 
- [DEFAULT]            MCU_FLAG_NAME = mmcu 
- [USER]               CFLAGS_STD = -std=gnu11 
- [USER]               CXXFLAGS_STD = -std=gnu++11 
- [COMPUTED]           DEVICE_PATH = /dev/cu.usbmodem* (from MONITOR_PORT)
- [DEFAULT]            FORCE_MONITOR_PORT =  
- [AUTODETECTED]       Size utility: AVR-aware for enhanced output
-
-                      ARDUINO_LIBS =
- [USER]                 Adafruit_PWM_Servo_Driver_Library
- [PLATFORM]             Wire
- [COMPUTED]           BOOTLOADER_PARENT = /Applications/Arduino.app/Contents/Java/hardware/arduino/avr/bootloaders (from ARDUINO_DIR)
- [COMPUTED]           ARDMK_VERSION = 1.5 
- [COMPUTED]           CC_VERSION = 9.1.0 (avr-gcc)
-------------------------
mkdir -p /home/x/Projects/y/z/software/arduino_uno/dragonfly//bin/uno/dragonfly
/usr/local/bin/avr-g++ -x c++ -include Arduino.h -MMD -c -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=189 -DARDUINO_ARCH_AVR -D__PROG_TYPES_COMPAT__ -I/Applications/Arduino.app/Contents/Java/hardware/arduino/avr/cores/arduino -I/Applications/Arduino
.app/Contents/Java/hardware/arduino/avr/variants/standard    -I/Applications/Arduino.app/Contents/Java/hardware/arduino/avr/libraries/Wire/src   -I/home/x/Projects/y/z/software/arduino_uno/dragonfly//lib/Adafruit_PWM_Servo_Driver_
Library -Wall -ffunction-sections -fdata-sections -Os -pedantic -Wall -Wextra -fpermissive -fno-exceptions -std=gnu++11 dragonfly.ino -o /home/x/Projects/y/z/software/arduino_uno/dragonfly//bin/uno/dragonfly/dragonfly.ino.o
/usr/local/bin/avr-g++ -MMD -c -mmcu=atmega328p -DF_CPU=16000000L -DARDUINO=189 -DARDUINO_ARCH_AVR -D__PROG_TYPES_COMPAT__ -I/Applications/Arduino.app/Contents/Java/hardware/arduino/avr/cores/arduino -I/Applications/Arduino.app/Contents/Java/hardwar
e/arduino/avr/variants/standard    -I/Applications/Arduino.app/Contents/Java/hardware/arduino/avr/libraries/Wire/src   -I/home/x/Projects/y/z/software/arduino_uno/dragonfly//lib/Adafruit_PWM_Servo_Driver_Library -Wall -ffunction-s
ections -fdata-sections -Os -pedantic -Wall -Wextra -fpermissive -fno-exceptions -std=gnu++11 /home/x/Projects/y/z/software/arduino_uno/dragonfly//lib/Adafruit_PWM_Servo_Driver_Library/Adafruit_PWMServoDriver.cpp -o /home/x/Pr
ojects/y/z/software/arduino_uno/dragonfly//bin/uno/dragonfly/userlibs/Adafruit_PWM_Servo_Driver_Library/Adafruit_PWMServoDriver.cpp.o
/home/x/Projects/y/z/software/arduino_uno/dragonfly//lib/Adafruit_PWM_Servo_Driver_Library/Adafruit_PWMServoDriver.cpp: In member function 'uint8_t Adafruit_PWMServoDriver::getPWM(uint8_t)':
/home/x/Projects/y/z/software/arduino_uno/dragonfly//lib/Adafruit_PWM_Servo_Driver_Library/Adafruit_PWMServoDriver.cpp:167:71: error: call of overloaded 'requestFrom(uint8_t, int, uint8_t)' is ambiguous
  167 |   _i2c->requestFrom((uint8_t)_i2caddr, LED0_ON_L + 4 * num, (uint8_t)4);
      |                                                                       ^
In file included from /home/x/Projects/y/z/software/arduino_uno/dragonfly//lib/Adafruit_PWM_Servo_Driver_Library/Adafruit_PWMServoDriver.h:27,
                 from /home/x/Projects/y/z/software/arduino_uno/dragonfly//lib/Adafruit_PWM_Servo_Driver_Library/Adafruit_PWMServoDriver.cpp:30:
/Applications/Arduino.app/Contents/Java/hardware/arduino/avr/libraries/Wire/src/Wire.h:62:13: note: candidate: 'uint8_t TwoWire::requestFrom(uint8_t, uint8_t, uint8_t)'
   62 |     uint8_t requestFrom(uint8_t, uint8_t, uint8_t);
      |             ^~~~~~~~~~~
/Applications/Arduino.app/Contents/Java/hardware/arduino/avr/libraries/Wire/src/Wire.h:65:13: note: candidate: 'uint8_t TwoWire::requestFrom(int, int, int)'
   65 |     uint8_t requestFrom(int, int, int);
      |             ^~~~~~~~~~~
make: *** [/home/x/Projects/y/z/software/arduino_uno/dragonfly//bin/uno/dragonfly/userlibs/Adafruit_PWM_Servo_Driver_Library/Adafruit_PWMServoDriver.cpp.o] Error 1
```

--- a/Adafruit_PWMServoDriver.cpp
+++ b/Adafruit_PWMServoDriver.cpp
@@ -46,7 +46,7 @@ Adafruit_PWMServoDriver::Adafruit_PWMServoDriver(TwoWire *i2c, uint8_t addr) {
  *  @brief  Setups the I2C interface and hardware
  *  @param  prescale
  *          Sets External Clock (Optional)
- *          
+ *
  */
 void Adafruit_PWMServoDriver::begin(uint8_t prescale) {
   _i2c->begin();
@@ -164,7 +164,7 @@ void Adafruit_PWMServoDriver::setPWMFreq(float freq) {
  *  @return requested PWM output value
  */
 uint8_t Adafruit_PWMServoDriver::getPWM(uint8_t num) {
-  _i2c->requestFrom((uint8_t)_i2caddr, LED0_ON_L + 4 * num, (uint8_t)4);
+  _i2c->requestFrom((int)_i2caddr, LED0_ON_L + 4 * num, (int)4);
   return _i2c->read();
 }
 

